source/data-api/generated-endpoints.txt

```diff
@@ -530,6 +629,95 @@ authentication providers:
              "collection": "hello",
            }'
 
+.. _data-api-authorize:
+
+Authorize the Request
+~~~~~~~~~~~~~~~~~~~~~
+
+Depending on your Data API :ref:`authorization configuration
+<data-api-authorization>`, your requests may need to include additional
+authorization information.
+
+.. tabs::
+     
+   .. tab:: No Additional Authorization
+      :tabid: no-authorization
+   
+      All authenticated users are authorized to call the endpoint.
+      Authenticated requests do not need to include authorization
+      information.
+      
+   .. tab:: Secret Parameter
+      :tabid: secret-parameter
+      
+      Authenticated users must prove that they are authorized to call
+      the endpoint by including the endpoint's secret string as the
+      value of the ``secret`` query parameter.
+      
+      .. example::
+         
+         The following ``curl`` request uses secret query parameter
+         validation with the secret string ``"12345"``:
+      
+         .. code-block:: bash
+            :emphasize-lines: 4
+      
+            curl -X POST \
+              https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/findOne?secret=12345 \
+              -H "Content-Type: application/json" \
+              -d '{
+                "dataSource": "mongodb-atlas",
+                "database": "learn-data-api",
+                "collection": "tasks",
+                "filter": { "text": "Do the dishes" }
+              }'
+
+   .. tab:: Payload Signature
+      :tabid: payload-signature
+   
+      Authenticated users must prove that they are authorized to call
+      the endpoint by including an ``Endpoint-Signature`` header that
+      contains a hexadecimal-encoded :abbr:`HMAC (Hash-based Message
+      Authentication Code)` SHA-256 hash generated from the request body
+      and the endpoint's secret string.
+
+      .. code-block:: none
+         
+         Endpoint-Signature::sha256=<hex encoded hash>
+      
+      You could use the following function to generate the payload
+      signature:
+      
+      .. code-block:: javascript
+      
+         // Generate an HMAC request signature
+         exports = function signEndpointRequest(
+           secret, // the secret validation string e.g. 12345
+           body, // the endpoint request body e.g. { "message": "MESSAGE" }
```

Comment:
```suggestion
         exports = function signEndpointRequest(
           secret, // the secret validation string e.g. 12345
           body, // the endpoint request body e.g. { "message": "MESSAGE" }
```

[suggestion] i think it'd be preferable to use JSDoc-style comments to describe these input params. more readable/industry standard.

smthn like:
```
/**
 * Add two numbers.
 * @param {string} secret - the secret validation string
 * @param {object} body - the endpoint request body e.g. { "message": "MESSAGE" }
 * @returns {?not sure?}
 */